src/VisualStudio/Core/Impl/Options/GridOptionPreviewControl.xaml.cs

```diff
@@ -93,5 +106,264 @@ internal override void Close()
+            };
+            if (sfd.ShowDialog() == System.Windows.Forms.DialogResult.OK)
+            {
+                IOUtilities.PerformIO(() =>
+                {
+                    var path = sfd.FileName;
+                    using (var sw = new StreamWriter(File.Create(path)))
+                    {
+                        sw.Write(editorconfig);
+                        sw.Close();
+                    }
+                });
+            }
+        }
+
+        internal static void Generate_Editorconfig(
+            OptionSet optionSet,
+            string language,
+            StringBuilder editorconfig,
+            Action<OptionSet, StringBuilder> getLangaugeSpecificEditorConfigOptions)
+        {
+            GenerateEditorconfig_CoreSettings(optionSet, language, editorconfig);
+            GenerateEditorconfig_DotNetSettings(optionSet, language, editorconfig);
+
+            getLangaugeSpecificEditorConfigOptions(optionSet, editorconfig);
+        }
+
+        private static void GenerateEditorconfig_CoreSettings(OptionSet optionSet, string language, StringBuilder editorconfig)
+        {
+            editorconfig.AppendLine("# Core EditorConfig Options");
+            editorconfig.AppendLine("# Comment the line below if you want to inherit parent .editorconfig settings.");
+            editorconfig.AppendLine("root = true");
+
+            editorconfig.AppendLine();
+            if (language == LanguageNames.CSharp)
+            {
+                editorconfig.AppendLine("# C# files");
+                editorconfig.AppendLine("[*.cs]");
+            }
+            else if (language == LanguageNames.VisualBasic)
+            {
+                editorconfig.AppendLine("# Basic files");
+                editorconfig.AppendLine("[*.vb]");
+            }
+
+            // indent_style
+            CoreCodeStyleOptions_GenerateEditorconfig(optionSet, FormattingOptions.UseTabs, language, editorconfig);
+            // indent_size
+            CoreCodeStyleOptions_GenerateEditorconfig(optionSet, FormattingOptions.IndentationSize, language, editorconfig);
+            // insert_final_newline
+            CoreCodeStyleOptions_GenerateEditorconfig(optionSet, FormattingOptions.InsertFinalNewLine, editorconfig);
+        }
+
+        private static void CoreCodeStyleOptions_GenerateEditorconfig(OptionSet optionSet, PerLanguageOption<bool> option, string language, StringBuilder editorconfig)
+        {
+            var element = option.StorageLocations.OfType<EditorConfigStorageLocation<bool>>().FirstOrDefault();
+            if (element != null)
+            {
+                editorconfig.Append(element.KeyName + " = ");
+                var curSetting = optionSet.GetOption(option, language);
+                if (curSetting)
+                {
+                    editorconfig.AppendLine("tab");
+                }
+                else
+                {
+                    editorconfig.AppendLine("space");
```

Comment:
this method seems really unintuitive. I would expect it to work for any `PerLanguageOption<bool>`, not just tab/spaces preference, but I don't know how the serialization of these works... maybe just consider  renaming the method to make this super clear?